src/main/frontend/db/model.cljs

```diff
@@ -1094,6 +1094,47 @@
       (util/safe-page-name-sanity-lc page))
      (distinct))))
 
+; return list of parents namespace
+(defn get-parents-namespace-list [page-namespace & nested-found]
+  (if (string/includes? page-namespace "/") ;isNamespace
```

Comment:
We should move any references to "/" outside fns to a separate `(def namespace-char "/")`. I'm surprised we don't already have this var defined elsewhere. You can create a regex dynamically with `(re-pattern namespace-char)`

Reply:
> I made some attempts in the `get-pages-relation` query with `not-join` but failed.. So the filter solution is good to me rofl

> Thanks for improving this. I'm also concerned about the complexity of maintaining this.

hello @cnrpman  and @logseq-cldwalker, thank you for giving me your time.
we all know that the solution I have implemented is **not the best** but with the `:block/ref` that are now available filtering them is the **only** solution.

if you agree I open a "discussions" to propose my idea on how to solve this problem.

meanwhile I continue with this solution...

Reply:
Can you use this?
https://github.com/logseq/logseq/blob/70acbadc70ef567ad96c63513a406a7118588d45/deps/graph-parser/src/logseq/graph_parser/text.cljs#L217